controllers/matryoshka/internal/kubeapiserver/kubeapiserver.go

```diff
@@ -256,6 +256,10 @@ func (r *Resolver) apiServerVolumeMounts(server *matryoshkav1alpha1.KubeAPIServe
 }
 
 func (r *Resolver) apiServerCommand(server *matryoshkav1alpha1.KubeAPIServer) []string {
+	featureGates := []string{}
```

Comment:
Two things here:
1. Usually, you don't use `[]<type>` to initialize slices but rather the zero value `var []<type>`
2. The `featureGates` should be allocated with capacity equal to the length of the feature gates.